test/integration/failover_stateful_test.lua

```diff
@@ -66,36 +50,155 @@ g.before_all(function()
         local vars = require('cartridge.vars').new('cartridge.roles.coordinator')
         vars.options.IMMUNITY_TIMEOUT = 0
+        vars.options.RECONNECT_PERIOD = 1
+        require('log').info('Coordinator options updated')
     ]])
-    g.cluster.main_server.net_box:call(
+end
+
+g_stateboard.before_all(function()
+    local g = g_stateboard
+    g.datadir = fio.tempdir()
+
+    fio.mktree(fio.pathjoin(g.datadir, 'stateboard'))
+    g.kvpassword = require('digest').urandom(6):hex()
+    g.state_provider = require('luatest.server'):new({
+        command = helpers.entrypoint('srv_stateboard'),
+        workdir = fio.pathjoin(g.datadir, 'stateboard'),
+        net_box_port = 14401,
+        net_box_credentials = {
+            user = 'client',
+            password = g.kvpassword,
+        },
+        env = {
+            TARANTOOL_LOCK_DELAY = 1,
+            TARANTOOL_PASSWORD = g.kvpassword,
+        },
+    })
+
+    g.state_provider:start()
+    g.client = stateboard_client.new({
+        uri = 'localhost:' .. g.state_provider.net_box_port,
+        password = g.kvpassword,
+        call_timeout = 1,
+    })
+
+    setup_cluster(g)
+
+    t.assert(g.cluster.main_server.net_box:call(
         'package.loaded.cartridge.failover_set_params',
         {{
             mode = 'stateful',
             state_provider = 'tarantool',
             tarantool_params = {
-                uri = g.stateboard.net_box_uri,
+                uri = g.state_provider.net_box_uri,
                 password = g.kvpassword,
             },
         }}
-    )
-    helpers.retrying({}, function()
-        g.stateboard:connect_net_box()
-        t.assert_covers(
-            g.stateboard.net_box:call('get_leaders'),
-            {[storage_uuid] = storage_1_uuid}
-        )
-    end)
+    ))
+end)
+
+g_etcd2.before_all(function()
+    local g = g_etcd2
+    local etcd_path = os.getenv('ETCD_PATH')
+    t.skip_if(etcd_path == nil, 'etcd missing')
+
+    local URI = 'http://127.0.0.1:14001'
+    g.datadir = fio.tempdir()
+    g.state_provider = {
+        workdir = fio.pathjoin(g.datadir, 'etcd2'),
+    }
+    function g.state_provider:start()
+        self.process = t.Process:start(etcd_path, {
+            '--data-dir', self.workdir,
+            '--listen-peer-urls', 'http://localhost:17001',
+            '--listen-client-urls', URI,
+            '--advertise-client-urls', URI,
+        })
+
+        helpers.retrying({}, function()
+            local resp = httpc.put(URI .. '/v2/keys/hello?value=world')
+            assert(resp.status == 200, resp.body)
+            require('log').info(
+                httpc.get(URI ..'/version').body
+            )
+        end)
+    end
+    function g.state_provider:stop()
+        local process = self.process
+        if process == nil then
+            return
+        end
+        self.process:kill()
+        helpers.retrying({}, function()
+            t.assert_not(process:is_alive(), 'Etcd is still running')
```

Comment:
IMO you shouldn't fail after timeout. Just send SIGKILL if it's still alive.

Reply:
I don't want to:

1. It's copy-pasted from test helpers
2. There's no convenient way to do it.